### PR TITLE
Avoid span conversion in delegate extension receiver

### PR DESCRIPTION
**Pull request by jjonescz**
> An extension method `Cᵢ.Mₑ` is ***eligible*** if:
> 
> - `Cᵢ` is a non-generic, non-nested class
> - The name of `Mₑ` is *identifier*
> - `Mₑ` is accessible and applicable when applied to the arguments as a static method as shown above
> - An implicit identity, reference ~~or boxing~~ **, boxing, or span** conversion exists from *expr* to the type of the first parameter of `Mₑ`.
>   👉 **Span conversion is not considered when overload resolution is performed for a method group conversion.**

Speclet updates: https://github.com/dotnet/csharplang/pull/8221, https://github.com/dotnet/csharplang/pull/8287
Test plan: https://github.com/dotnet/roslyn/issues/73445

### CODE REVIEW

**Comment by jjonescz** [comment]
@cston @333fred for reviews, thanks.

**Comment by cston** [comment]
It looks like there is still an open question in the speclet about whether to take the breaking change or not. Was that discussed in LDM?

**Comment by jjonescz** [comment]
> It looks like there is still an open question in the speclet about whether to take the breaking change or not. Was that discussed in LDM?

Not yet, I'm optimistically assuming we won't take a breaking change here. But if we will, we can always revert this. Or we can postpone this PR until after LDM if you prefer.

**Comment by cston** [comment]
> Or we can postpone this PR until after LDM if you prefer.

Is this change blocking other work? If not, I think my preference is to postpone this PR until after an LDM decision.

**Comment by jjonescz** [comment]
@cston @333fred this is ready for reviews again, thanks

File: src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs [hunk 1]
```diff
@@ -1891,16 +1891,16 @@ private static bool HasIdentityConversionToAny(NamedTypeSymbol type, ArrayBuilde
...
             Debug.Assert((object)thisType != null);
-            var conversion = this.ClassifyImplicitExtensionMethodThisArgConversion(sourceExpressionOpt: null, thisType, parameterType, ref useSiteInfo);
+            var conversion = this.ClassifyImplicitExtensionMethodThisArgConversion(sourceExpressionOpt: null, thisType, parameterType, ref useSiteInfo, isMethodGroupConversion);
             return IsValidExtensionMethodThisArgConversion(conversion) ? conversion : Conversion.NoConversion;
         }
 
         // Spec 7.6.5.2: "An extension method ... is eligible if ... [an] implicit identity, reference,
         // or boxing conversion exists from expr to the type of the first parameter"
-        public Conversion ClassifyImplicitExtensionMethodThisArgConversion(BoundExpression sourceExpressionOpt, TypeSymbol sourceType, TypeSymbol destination, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+        public Conversion ClassifyImplicitExtensionMethodThisArgConversion(BoundExpression sourceExpressionOpt, TypeSymbol sourceType, TypeSymbol destination, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo, bool isMethodGroupConversion)
```
Comment (333fred) [inline src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs:1903]: Should we update the comment above this method?

**Review by 333fred — APPROVED** [review]
LGTM (commit 2)

**Comment by jjonescz** [comment]
@cston for the second review, thanks

**Review by cston — APPROVED** [review]

**Pull request merged**